telemetry/server.go

```diff
@@ -345,8 +189,8 @@ func (s *Server) createProtocolStats(w http.ResponseWriter, r *http.Request) {
 	}
 	defer r.Body.Close()
 
-	peerIDHash := sha256.Sum256([]byte(protocolStats.PeerID))
-	protocolStats.PeerID = hex.EncodeToString(peerIDHash[:])
+	peerIDHash := sha256.Sum256([]byte(protocolStats.data.PeerID))
+	protocolStats.data.PeerID = hex.EncodeToString(peerIDHash[:])
```

Comment:
Remembering past discussions, maybe we should just store the peerID without hashing it? wdyt?

Reply:
I did not want to change it here to avoid bringing undrelated changes, but can do if it is pereferred:)

Reply:
Can be done either on a diff PR or in a diff commit! no preferences here either :rocket: